mkosi/__init__.py

```diff
@@ -721,9 +618,11 @@ def create_image(args: CommandLineArguments, for_cache: bool) -> Optional[Binary
         )
         output.append(f)
         disable_cow(f.name)
-        f.truncate(image_size(args))
+        disk_size = args.partition_table.disk_size()
+        f.truncate(disk_size)
 
-        exec_sfdisk(args, f)
+        if args.partition_table.partitions:
```

Comment:
maybe move this check into run_sfdisk()?

Reply:
I thought about this, but decided to use the current form. In principle you can execute sfdisk even without any partitions, and it's a completely valid and meaningful operation. We don't want to do this here, because we only interact with the disk through partitions, so we only want to format in that case.